src/components/TradeAssetSearch/hooks/useGetCustomTokensQuery.tsx

```diff
@@ -0,0 +1,61 @@
+import { type ChainId } from '@shapeshiftoss/caip'
+import { useQueries, type UseQueryResult } from '@tanstack/react-query'
+import type { TokenMetadataResponse } from 'alchemy-sdk'
+import { useCallback, useMemo } from 'react'
+import { isAddress } from 'viem'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { getAlchemyInstanceByChainId } from 'lib/alchemySdkInstance'
+
+type TokenMetadata = TokenMetadataResponse & {
+  chainId: ChainId
+  contractAddress: string
+  price: string
+}
+
+type UseGetCustomTokensQueryProps = {
+  contractAddress: string
+  chainIds: ChainId[]
+}
+
+type UseGetCustomTokenQueryReturn = {
+  data: UseQueryResult<TokenMetadata | null, Error>[]
+  isLoading: boolean
+}
+
+export const useGetCustomTokensQuery = ({
+  contractAddress,
+  chainIds,
+}: UseGetCustomTokensQueryProps): UseGetCustomTokenQueryReturn => {
+  const customTokenImportEnabled = useFeatureFlag('CustomTokenImport')
+
+  const queryKey = useMemo(() => ['customTokens', contractAddress], [contractAddress])
+
+  const getTokenMetadata = useCallback(
+    async (chainId: ChainId) => {
+      const alchemy = getAlchemyInstanceByChainId(chainId)
+      const tokenMetadataResponse = await alchemy.core.getTokenMetadata(contractAddress)
+      // TODO: get price from somewhere
+      return { ...tokenMetadataResponse, chainId, contractAddress, price: '0' }
+    },
+    [contractAddress],
+  )
+
+  const queryFn = useCallback(
+    (chainId: ChainId) =>
+      isAddress(contractAddress, { strict: false }) ? getTokenMetadata(chainId) : null,
+    [contractAddress, getTokenMetadata],
+  )
+
+  const customTokensQuery = useQueries({
+    queries: chainIds.map(chainId => ({
+      queryKey: [queryKey, chainId],
```

Comment:
preferably blocking: With this, prefer we take a similar approach to the rFOX hooks where we have a `getQueryKey` which returns an array, in this case `['customTokens', contractAddress, chainId]` instead of `[['customTokens', contractAddress], chainId]`

Reply:
https://github.com/shapeshift/web/pull/7285/commits/3ef813b0052ecf929ba93b57260242ce59c7debf